.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+repos:
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: "v2.6.2"
+    hooks:
+      - id: prettier
```

Comment:
Just to confirm, does this trigger our prettier config? I believe we could have a one commit hook that will run `yarn format`.

Reply:
Maybe we could create our custom hook eg:
```
- id: format
  name: "code formatting"
  entry: yarn format
  description: "Checks files formatting with prettier"
```

Reply:
That's a good point. Indeed, pre-commit hooks aren't run with `.eslintrc.json` and `.prettierrc.json` configuration without an additional configuration like creating custom hooks.

It seems like using [Husky](https://typicode.github.io/husky/#/) makes easier to run git hooks using Yarn scripts. See https://github.com/threshold-network/subgraph/pull/14/commits/52edfec8ab7dba6fb5cf503092326c79e7d8cb6b